src/syscall/syscall_nonlibc.go

```diff
@@ -0,0 +1,7 @@
+// +build !darwin,!nintendoswitch,!wasi
```

Comment:
When is this needed? Isn't this just the baremetal case? In that case, I think this file should be removed and `Getenv` should be moved back to syscall_baremetal.go.